Blish HUD/GameServices/Input/Keyboard/TypedInputUtil.cs

```diff
@@ -151,5 +170,28 @@ internal static class TypedInputUtil {
             } while (rc < 0);
         }
 
+        // Reference: https://stackoverflow.com/a/17387605
+        public static char GetModifiedKey(char c) {
+            short vkKeyScanResult = VkKeyScan(c);
+
+            // a result of -1 indicates no key translates to input character
+            if (vkKeyScanResult == -1)
+                throw new ArgumentException("No key mapping for " + c);
+
+            // vkKeyScanResult & 0xff is the base key, without any modifiers
+            uint code = (uint)vkKeyScanResult & 0xff;
+
+            // set shift key pressed
+            byte[] b = new byte[256];
+            b[0x10] = 0x80;
+
+            uint r;
+            // return value of 1 expected (1 character copied to r)
+            if (1 != ToAscii(code, code, b, out r, 0))
```

Comment:
Worth trying ToUnicode to reflect the old implementation.

Reply:
I would like to revisit the whole keyboard handling in the near future but I think it's more important to have a quick solution for now that is working since the shift key is't working at all and somehow important for the functionality of at least the KillProof module.

Reply:
I was trying out your fork and it appears sometimes the bug that the shift toggle happens at the second or third letter still appears. There seems to be still something off with the state.

Reply:
I think I've found what's causing that. I'm currently reworking the whole function to be more rubust and use `ToUnicodeEx`.
Please consider my PR as on-hold until the next commit.

Reply:
I just noticed that accent and diacritic characters aren't working in either implementation so I'll fix that too.

Reply:
So I've made some modifications but as @greaka pointed out setting the key state for shift manually is not the mose elegant solution and I agree.
However, the problem seems that using `GetKeyboardState` and passing it directly to `ToUnicode` isn't working because it seems to set the low-order bit for the keys which is for 'toggle keys' whereas `ToUnicode` expects the high-oder bit set.
Here is what I've tried:
```cs
byte[] keyState = new byte[256];
GetKeyboardState(keyState);
ToUnicode(virtKeyCode, scanCode, keyState, output, (int)5, (uint)0);
```
Even getting some key states before calling `GetKeyboardState` as mentioned here ( https://stackoverflow.com/a/53713024 ) does not work (for me at least).

Reply:
New insights:
Adding `Keys.(Left|Right)Shift` and `Keys.(Left|Right)Control` to `_hookIgnoredKeys` is necessary for the systems `User32.dll` to know of the states of these keys.
`GetKeyboardState` will then work if a `GetKeyState` is called before as stated in the SO post above.

This leaves us with all modified keys working.

Dead key implementation still pending.

Reply:
Tested. The bug that caps begins to apply to the second or third letter appear to be gone. 🚀

However, the current implementation seems to be blocking the key-combos in the [TextInputBase](https://github.com/blish-hud/Blish-HUD/blob/0e82ee8668b1e8fcb835795c4735e1ab360947e9/Blish%20HUD/Controls/TextInputBase.cs#L214-L243) or that control base has to be updated. Mainly I tried Ctrl+A (select all), Ctrl+C (copy), Ctrl+V (paste)
It could be an easiy fix though.